src/main/java/uk/gov/hmcts/darts/audio/component/impl/AudioMessageDigestImpl.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.darts.audio.component.impl;
+
+import uk.gov.hmcts.darts.audio.component.AudioMessageDigest;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+
+import java.security.NoSuchAlgorithmException;
+
+import static uk.gov.hmcts.darts.audio.exception.AudioApiError.FAILED_TO_UPLOAD_AUDIO_FILE;
+
+public class AudioMessageDigestImpl implements AudioMessageDigest {
+
+    private final String algorithm;
+
+    public AudioMessageDigestImpl(String algorithm) {
+        this.algorithm = algorithm;
+    }
+
+    @Override
+    public java.security.MessageDigest getMessageDigest() {
+        java.security.MessageDigest digest;
+        try {
+            digest = java.security.MessageDigest.getInstance(algorithm);
+        } catch (NoSuchAlgorithmException e) {
+            throw new DartsApiException(FAILED_TO_UPLOAD_AUDIO_FILE, e);
+        }
+        return digest;
+    }
```

Comment:
Is it necessary to fully-qualify the `java.security.MessageDigest` class name?

Reply:
I will correct